crates/holochain/src/core/ribosome.rs

```diff
@@ -370,6 +430,14 @@ pub struct ZomeCallInvocation {
     /// The provenance of the call. Provenance means the 'source'
     /// so this expects the `AgentPubKey` of the agent calling the Zome function
     pub provenance: AgentPubKey,
+    /// The signature of the call from the provenance of the call.
+    /// Everything except the signature itself is signed.
+    pub signature: Signature,
```

Comment:
Kind of weird that the signature is included along that which is signed. How do you get from this to "everything but the signature"?

Reply:
@maackle yeah let's roll with what we have for now, and if/when we refactor it i think we'll go even deeper and convert the signed data to opaque bytes as per @neonphog 's earlier suggestions